metis-processor/src/main/java/eu/europeana/metis/processor/utilities/ImageEnhancerUtil.java

```diff
@@ -13,227 +16,338 @@
-                LOGGER.debug("{}\t=>\t{}", thumbnailResource.getAbout(), thumbnail.getTargetName());
-//                Files.write(Paths.get("/tmp/test_thumbnails/newThumbnail-" + thumbnail.getTargetName()), thumbnail.getContentStream().readAllBytes());
-                s3Client.putIbmObject(thumbnail.getTargetName(), thumbnail.getContentStream(), prepareObjectMetadata(thumbnailResource.getMimeType(), thumbnail.getContentSize()));
-            }
+          if (ArrayUtils.isEmpty(largeThumbnail)) {
+            largeThumbnail = s3Client.getAmazonObject(largeThumbnailObjectName);
+          }
+          if (isWebResourceResolutionAndImageResolutionSame(webResourceWrapper, largeThumbnail)) {
+            //                        Files.write(Paths.get("/tmp/test_thumbnails/" + largeThumbnailObjectName), largeThumbnail);
+            final long startTime = System.nanoTime();
+            byte[] enhancedImage = enhanceImage(largeThumbnail, largeThumbnailObjectName);
+            //                        Files.write(Paths.get("/tmp/test_thumbnails/enhanced-" + largeThumbnailObjectName), enhancedImage);
+            final long elapsedTimeInNanoSec = System.nanoTime() - startTime;
+
+            List<Thumbnail> thumbnails = generateThumbnails(webResourceWrapper, largeThumbnailObjectName,
+                enhancedImage);
+            uploadThumbnails(webResourceWrapper, thumbnails);
+            appendSuccessReport(webResourceWrapper, enhancedImage, thumbnailHex, elapsedTimeInNanoSec, reportRow);
+          } else {
+            appendProcessedReport(webResourceWrapper, thumbnailHex, reportRow);
+          }
+        } else {
+          appendUnProcessedReport(webResourceWrapper, "", reportRow);
         }
+      } catch (MediaExtractionException | IOException e) {
+        LOGGER.error("Failed enhancing thumbnail image {} {}", webResourceWrapper.getAbout(), e);
+        appendFailReport(reportRow);
+      }
+    } else {
+      try {
+        appendDryRunReport(webResourceWrapper, md5Hex(webResourceWrapper.getAbout()), reportRow);
+      } catch (MediaExtractionException e) {
+        LOGGER.error("Failed dry run {} {}", webResourceWrapper.getAbout(), e);
+        appendFailReport(reportRow);
+      }
     }
+  }
 
-    private void appendSuccessReport(WebResourceWrapper thumbnailResource, byte[] enhancedImage, String thumbnailHex, long elapsedTimeInNanoSec, ReportRow reportRow) {
-        reportRow.setHeightBefore(thumbnailResource.getHeight());
-        reportRow.setWidthBefore(thumbnailResource.getWidth());
-        reportRow.setElapsedTime(TimeUnit.MILLISECONDS.convert(elapsedTimeInNanoSec, TimeUnit.NANOSECONDS));
-        reportRow.setImageLinkHex(thumbnailHex);
-        appendNewWidthAndHeight(enhancedImage, reportRow);
-        reportRow.setStatus("SUCCESS");
-        fileCsvImageReporter.appendRow(reportRow);
-    }
+  private void processShownBy(RDF recordToProcess, Aggregation aggregationItem, String recordId) {
```

Comment:
Isn't this method and the processHasView practically identical and can therefore be merged a single generic one?

Reply:
:thinking: . true. nice catch. done.